src/translation.h

```diff
@@ -23,7 +23,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include <string>
 
 class Translations;
-extern Translations *g_translations;
```

Comment:
I think the client should rather keep its own `g_client_translations` and the `g_all_translations` should only be used for the server.

Reply:
Yes, the problem is `g_*_translations` is used once deep in recursive `translate_string` function. The trick of using a "" lang code (or the appropriate code) for client make this function work for both client and server.
So having a separate `g_client_translations` would imply to add something like an extra bool argument to this function to tell if we are on a server or client side translation. Or use a specific hardcoded lang_code for client (like "client" ?). I'm not sure which of these (including keeping as it is) is better.
By the way, choice made would not affect much code outside string.cpp

Reply:
> like an extra bool argument to this function

Or better pass a `const Translations*`. If there ever is some cleanup to remove the global variable for the server, that will come in useful.

Reply:
.. instead of lang_code, sounds good !